core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -405,6 +411,7 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
                     b.bind(IndicesQueriesRegistry.class).toInstance(searchModule.getQueryParserRegistry());
                     b.bind(SearchRequestParsers.class).toInstance(searchModule.getSearchRequestParsers());
                     b.bind(SearchExtRegistry.class).toInstance(searchModule.getSearchExtRegistry());
+                    b.bind(NamedXContentRegistry.class).toInstance(xContentRegistry);
```

Comment:
I hate binding another thing but in subsequent PRs I'll use this to remove a bunch of things.